devops/scripts/letsencrypt.sh

```diff
@@ -0,0 +1,111 @@
+#!/bin/bash
+set -e
+
+script_dir=$(realpath $(dirname "${BASH_SOURCE[0]}"))
+
+if [[ -z ${STORAGE_ACCOUNT} ]]; then
+  echo "STORAGE_ACCOUNT not set"
+  exit 1
+fi
+
+staticenabled=$(az storage blob service-properties show -o tsv \
+    --account-name "${STORAGE_ACCOUNT}" \
+    --auth-mode login \
+    --query 'staticWebsite.enabled')
+
+if [[ ${staticenabled} = 'false' ]]; then
+    # Enable static-website hosting
+    echo "Enabling static Web hosting on storage account"
+    az storage blob service-properties update \
+        --account-name "${STORAGE_ACCOUNT}" \
+        --auth-mode login \
+        --static-website \
+        --index-document 'index.html'
+else
+    echo "Static Web hosting already enabled on storage account"
+fi
+
+# Create the default index.html page
+cat << EOF > index.html
+<!DOCTYPE html>
+<html lang="en" xmlns="http://www.w3.org/1999/xhtml"><head><meta charset="utf-8"/><title></title></head><body></body></html>
+EOF
+
+echo "Checking for index.html file in storage account"
+
+isindex=$(az storage blob list -o json \
+    --account-name "${STORAGE_ACCOUNT}" \
+    --auth-mode login \
+    --container-name '$web' \
+    --query "[?name=='index.html'].name" \
+    | jq 'length')
+
+if [[ ${isindex} -lt 1 ]]; then
+    echo "Uploading index.html file"
+    # Upload the default index.html page
+    az storage blob upload \
+        --account-name "${STORAGE_ACCOUNT}" \
+        --auth-mode login \
+        --container-name '$web' \
+        --file index.html \
+        --name index.html \
+        --no-progress \
+        --only-show-errors
+
+    # Wait a bit for the App gateway health probe to notice.
+    echo "Waiting 30s for health probe"
```

Comment:
would be good to use the health probe interval parameter (times a factor), in case it changes. I understand it's not in the context of this script, but can be made available if we think it safer. Or do we have confident that 30s will always be enough?

Reply:
The [default probe](https://docs.microsoft.com/azure/application-gateway/application-gateway-probe-overview#default-health-probe) timing is 30sec and it only needs a single good response, so 30sec wait seems okay.